apps/server/src/modules/files-storage/controller/files-storage.consumer.spec.ts

```diff
@@ -0,0 +1,157 @@
+import { createMock, DeepMocked } from '@golevelup/ts-jest';
+import { MikroORM } from '@mikro-orm/core';
+import { ObjectId } from '@mikro-orm/mongodb';
+import { Test, TestingModule } from '@nestjs/testing';
+import { EntityId, FileRecordParentType } from '@shared/domain';
+import { courseFactory, fileRecordFactory, setupEntities } from '@shared/testing';
+import { Logger } from '@src/core/logger';
+import { FileRecordUC } from '../uc/file-record.uc';
+import { FilesStorageUC } from '../uc/files-storage.uc';
+import { FileRecordResponse } from './dto';
+import { FilesStorageConsumer } from './files-storage.consumer';
+
+describe('FilesStorageConsumer', () => {
+	let module: TestingModule;
+	let fileRecordUC: DeepMocked<FileRecordUC>;
+	let filesStorageService: DeepMocked<FilesStorageUC>;
+	let service: FilesStorageConsumer;
+	let orm: MikroORM;
+
+	afterEach(() => {
+		jest.clearAllMocks();
+	});
+
+	afterAll(async () => {
+		await orm.close();
+		await module.close();
+	});
+
+	beforeAll(async () => {
+		orm = await setupEntities();
+		module = await Test.createTestingModule({
+			providers: [
+				FilesStorageConsumer,
+				{
+					provide: FileRecordUC,
+					useValue: createMock<FileRecordUC>(),
+				},
+				{
+					provide: FilesStorageUC,
+					useValue: createMock<FilesStorageUC>(),
+				},
+				{
+					provide: Logger,
+					useValue: createMock<Logger>(),
+				},
+				{
+					provide: MikroORM,
+					useValue: orm,
+				},
+			],
+		}).compile();
+
+		fileRecordUC = module.get(FileRecordUC);
+		filesStorageService = module.get(FilesStorageUC);
+		service = module.get(FilesStorageConsumer);
+	});
+
+	describe('copyFilesOfParent()', () => {
+		const schoolId: EntityId = new ObjectId().toHexString();
+		it('should call filesStorageService.copyFilesOfParent with params', async () => {
+			const payload = {
+				userId: new ObjectId().toHexString(),
+				source: {
+					parentId: new ObjectId().toHexString(),
```

Comment:
A helper function that can be used for any source or target are helpful to cleanup the code.

```
createFileRecordParams(id?) {
const entity = courseFactory.buildWithId();
const param: FileRecordParams = { .. }

return param;
}
```
One more helper for build createCopyFilesOfSameParentPayload() : CopyFilesOfParentPayload